File: src/main/java/com/flowcommerce/core/storage/member/SellerHistoryEntity.java
package com.flowcommerce.core.storage.member;

import com.flowcommerce.core.domain.HistoryType;
import com.flowcommerce.core.domain.member.RejectionReason;
import com.flowcommerce.core.domain.member.SellerStatus;
import com.flowcommerce.core.storage.BaseEntity;
import jakarta.persistence.Entity;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class SellerHistoryEntity extends BaseEntity {
    private Long sellerId;
    private HistoryType type;
    private SellerStatus previousStatus;
    private SellerStatus newStatus;
    private RejectionReason rejectionReason;
    private String rejectionDetail;
    private Long processedBy;
}

File: src/main/kotlin/com/flowcommerce/core/storage/member/SellerHistoryEntity.kt
package com.flowcommerce.core.storage.member

import com.flowcommerce.core.domain.HistoryType
import com.flowcommerce.core.domain.member.RejectionReason
import com.flowcommerce.core.domain.member.SellerStatus
import com.flowcommerce.core.storage.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated

@Entity
class SellerHistoryEntity(
    var sellerId: Long?,

    @Enumerated(EnumType.STRING)
    var type: HistoryType?,

    @Enumerated(EnumType.STRING)
    var previousStatus: SellerStatus?,

    @Enumerated(EnumType.STRING)
    var newStatus: SellerStatus?,

    @Enumerated(EnumType.STRING)
    var rejectionReason: RejectionReason?,

    var rejectionDetail: String?,

    var processedBy: Long?
) : BaseEntity() {
    protected constructor() : this(null, null, null, null, null, null, null)
}
